src/Result.php

```diff
@@ -10,6 +10,15 @@ final class Result
 {
     private $valid = true;
     private $errors = [];
+    
+    public function __construct(iterable $messages = [])
```

Comment:
How would you use it?

Reply:
It can be something like this:

```php
$rules = new Rules([
    function ($value): Result {
        $errors =[] ;

        if ($value !== 41) {
            $errors[] = 'Value should be 41!';
        }
        if ($value !== 42) {
            $errors[] = 'Value should be 42!';
        }
        if ($value !== 43) {
            $errors[] = 'Value should be 43!';
        }

        return new Result($errors);
    }
]);
```

Reply:
Not a big gain compared to:

```php
$rules = new Rules([
    function ($value): Result {
       $result = new Result();

        if ($value['first'] === 41) {
            $result->addError('First should not be 41!');
        }
        if ($value['second'] === 42) {
            $result->addError('Second should not be 42!');
        }
        if ($value['third'] === 43) {
            $result->addError('Third should not be 43!');
        }

        return $result;
    }
]);
```

And having two ways of using it makes things a bit more complicated.